src/luos.c

```diff
@@ -724,143 +653,126 @@ error_return_t Luos_ReceiveData(container_t *container, msg_t *msg, void *bin_da
     }
     return FAILED;
 }
+
 /******************************************************************************
  * @brief Send datas of a streaming channel
  * @param Container who send
  * @param Message to send
  * @param streaming channel pointer
  * @return None
  ******************************************************************************/
-void Luos_SendStreaming(container_t *container, msg_t *msg, streaming_channel_t *stream)
-{
+void Luos_SendStreaming(container_t *container, msg_t *msg, streaming_channel_t *stream) {
     // Compute number of message needed to send available datas on ring buffer
     int msg_number = 1;
     int data_size = Stream_GetAvailableSampleNB(stream);
     const int max_data_msg_size = (MAX_DATA_MSG_SIZE / stream->data_size);
-    if (data_size > max_data_msg_size)
-    {
+    if (data_size > max_data_msg_size) {
         msg_number = (data_size / max_data_msg_size);
         msg_number += ((msg_number * max_data_msg_size) < data_size);
     }
 
     // Send messages one by one
-    for (volatile uint16_t chunk = 0; chunk < msg_number; chunk++)
-    {
+    for (volatile uint16_t chunk = 0; chunk < msg_number; chunk++) {
         // compute chunk size
-        uint16_t chunk_size = 0;
-        if (data_size > max_data_msg_size)
-        {
+        uint16_t chunk_size = data_size;
+        if (data_size > max_data_msg_size) {
             chunk_size = max_data_msg_size;
         }
-        else
-        {
-            chunk_size = data_size;
-        }
 
         // Copy data into message
         Stream_GetSample(stream, msg->data, chunk_size);
 
         // Send message
-        while (Luos_SendMsg(container, msg) == FAILED)
-        {
+        while (Luos_SendMsg(container, msg) == FAILED) {
             Luos_Loop();
         }
 
         // check end of data
-        if (data_size > max_data_msg_size)
-        {
+        if (data_size > max_data_msg_size) {
```

Comment:
As I understand `data_size` is a local variable check the end of data is useless. No message is returned neither value. `data_size` erased at the end of the function. The check make sens if the function return an action `error` or `status`.